app/src/main/java/org/lineageos/selfie/MainActivity.kt

```diff
@@ -717,6 +728,27 @@ class MainActivity : AppCompatActivity() {
         bindCameraUseCases()
     }
 
+    private fun cycleVideoQuality() {
+        if (!canRestartCamera()) {
+            return
+        }
+
+        val newVideoQuality = if (videoQuality == camera.supportedVideoQualities.first()) {
+            camera.supportedVideoQualities.last()
+        } else {
+            val index = camera.supportedVideoQualities.indexOf(videoQuality)
+            camera.supportedVideoQualities[maxOf(0, index - 1)]
```

Comment:
did maxOf() here just to be safe, idk if there are any cameras that only support single video recording resolution.

Reply:
Yep, fine